packages/eslint-plugin/src/rules/no-shadow.ts

```diff
@@ -99,7 +99,9 @@ export default util.createRule<Options, MessageIds>({
       }
 
       const isShadowedValue =
-        'isValueVariable' in shadowed ? shadowed.isValueVariable : true;
+        !('isValueVariable' in shadowed) ||
+        (shadowed.defs[0]?.type !== DefinitionType.ImportBinding &&
```

Comment:
this isn't quite correct because it will match all import variables.
We specifically only want to match variables imported via `import type`.

Reply:
Good catch! It should be fixed with the `fixup`.